google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractReadContext.java

```diff
@@ -419,10 +430,14 @@ void initTransaction() {
   }
 
   @Override
-  public void setSpan(Span span) {
+  public void setSpan(ISpan span) {
     this.span = span;
   }
 
+  /** No-op method needed to implement SessionTransaction interface. */
```

Comment:
Would it not make more sense to change the interface definition to accept an `ISpan` instead?

Reply:
Its a public method, changing this would result in breaking changes. So instead of changing I created a new overload with `ISpan` , please recommend if anything can be done here ?

Reply:
Yeah, that is a Java-strangeness. All methods in an interface are considered public. But if the interface itself is not public, then those methods are still not accessible from outside the package, even though they are marked as public. And in this case the interface `SessionTransaction` is package-private, which again means that we can safely change the API.

Reply:
Oh, that's an interesting find. `PartitionedDmlTransaction` should never have been `public` in the first place... AFAICT, this is also the only public implementation of this interface. And changing it to package-private works for the rest of the client library, so that is probably something that we should consider doing in a separate PR.

Reply:
Done